src/contracts/Staking.sol

```diff
@@ -172,7 +168,18 @@ contract Staking is OwnableUpgradeable, StakingStorage {
      */
     function setAffiliateFee(uint256 _affiliateFee) external onlyOwner {
         affiliateFee = _affiliateFee;
-        emit LogSetAffiliateFee(block.number, _affiliateFee);
+        emit LogSetAffiliateFee(_affiliateFee);
+    }
+
+    /**
+        @notice sets a limit to the amount of staking that can occur to allow for a guarded 
+        launch. This only applies to staking and will disable new stakers based on the total supply
+        of the paired Yieldy token. 
+        @param _totalSupplyLimit uint can be set to uint256.max to disable limits. 
+    */
+    function setTotalSupplyLimit(uint256 _totalSupplyLimit) external onlyOwner {
+        totalSupplyLimit = _totalSupplyLimit;
```

Comment:
For consistency sake should we check for MAX_SUPPLY here?

Reply:
Not sure I am following, the MAX_SUPPLY of what the yieldy?

Reply:
Yea, we have a `MAX_SUPPLY` of the yieldy.  It seems like a non issue, but just weird logic where totalSupplyLimit could be higher than MAX_SUPPLY.  If someone wanted to change these contracts for a more realistic MAX_SUPPLY const then totalSupplyLimit could potentially be much larger than MAX_SUPPLY, which with current logic wouldn't break anything.

Reply:
Got it.  I think its fine to leave as is. For example, I can see this being set to uin256.max to enable unlimited deposits by the TMDC without them having to worry about additional variables.  Your call.